### ISSUE TITLE
Update H2 to 1.4.199

### ISSUE THREAD

**Issue by axelfontaine** [flyway/flyway#2326]

**Comment by josephlbarnett** [comment]
1.4.199 is now the latest

**Comment by clemp6r** [comment]
Are you sure this works well with 1.4.199?
I have the following error (Spring Boot app startup):

```
Caused by: org.flywaydb.core.internal.exception.FlywaySqlException: 
Unable to check whether schema "PUBLIC" exists
----------------------------------------------
SQL State  : 42S02
Error Code : 42102
Message    : Table "schemata" non trouvée
Table "schemata" not found; SQL statement:
SELECT COUNT(*) FROM INFORMATION_SCHEMA.schemata WHERE schema_name=? [42102-199]

	at org.flywaydb.core.internal.database.base.Schema.exists(Schema.java:75)
	at org.flywaydb.core.internal.command.DbValidate.validate(DbValidate.java:108)
	at org.flywaydb.core.Flyway.doValidate(Flyway.java:1476)
	at org.flywaydb.core.Flyway.access$100(Flyway.java:85)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:1364)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:1356)
	at org.flywaydb.core.Flyway.execute(Flyway.java:1711)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:1356)
	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1837)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1774)
	... 119 common frames omitted
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "schemata" non trouvée
Table "schemata" not found; SQL statement:
SELECT COUNT(*) FROM INFORMATION_SCHEMA.schemata WHERE schema_name=? [42102-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.command.Parser.readTableOrView(Parser.java:7146)
	at org.h2.command.Parser.readTableFilter(Parser.java:1895)
	at org.h2.command.Parser.parseSelectSimpleFromPart(Parser.java:2641)
	at org.h2.command.Parser.parseSelectSimple(Parser.java:2788)
	at org.h2.command.Parser.parseSelectSub(Parser.java:2636)
	at org.h2.command.Parser.parseSelectUnion(Parser.java:2469)
	at org.h2.command.Parser.parseSelect(Parser.java:2440)
	at org.h2.command.Parser.parsePrepared(Parser.java:814)
	at org.h2.command.Parser.parse(Parser.java:788)
	at org.h2.command.Parser.parse(Parser.java:760)
	at org.h2.command.Parser.prepareCommand(Parser.java:683)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:77)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:349)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:311)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.prepareStatement(JdbcTemplate.java:333)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.queryForInt(JdbcTemplate.java:138)
	at org.flywaydb.core.internal.database.h2.H2Schema.doExists(H2Schema.java:48)
	at org.flywaydb.core.internal.database.base.Schema.exists(Schema.java:73)
	... 129 common frames omitted
```

**Comment by josephlbarnett** [comment]
https://github.com/flyway/flyway/pull/2325 in master should take care of that issue

**Comment by mdagcilar** [comment]
I am experiencing this issue with h2. Everything works fine with h2database version 1.4.197, but from 1.4.198 onwards i get this issue

```
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "schemata" not found; SQL statement:
SELECT COUNT(*) FROM INFORMATION_SCHEMA.schemata WHERE schema_name=? [42102-198]```

**Comment by hilmarf** [comment]
@mdagcilar - true, I'm also facing similar issue. Unfortunately Spring, h2, Flyway aren't very well aligned regarding naming conventions. One enforces all uppercase, the other everything lowercase and don't even think of mixed case.... where are we? In the 21st century and still having case issues? Why not keeping things as they are specified in the code? Let developers decide if they want to have all upper/lower/mixed case...

Thanks!

**Comment by axelfontaine** [comment]
As you can see in the details of this issue, this was fixed in 6.0.0-beta2.
